server/main.lua

```diff
@@ -351,6 +351,12 @@ QBCore.Functions.CreateCallback('qb-inventory:server:attemptPurchase', function(
         end
     end
 
+    if amount > itemInfo.amount then
```

Comment:
`itemInfo` is based on `data` which comes from the client through the NUI callback. Since that information already exists on the server, it'd be best to use that trusted server-side information to prevent client exploits.
```suggestion
    if shopInfo.items[itemInfo.slot].amount > itemInfo.amount then
```

Reply:
Hey, thank you for pointing this out. I agree with you, looks like I missed the `shopInfo` variable. I have updated my code.